python-libraries/nanover-core/src/nanover/utilities/timing.py

```diff
@@ -22,26 +22,22 @@ def interval(self, value):
             self._interval = value
 
     def yield_interval(self):
-        last_yield = time.monotonic() - self.interval
+        yield 0
         while True:
-            time_since_yield = time.monotonic() - last_yield
-            wait_duration = max(0.0, self.interval - time_since_yield)
-            time.sleep(wait_duration)
-            yield time.monotonic() - last_yield
             last_yield = time.monotonic()
+            time.sleep(self.interval)
```

Comment:
That's what I thought too but that isn't how it actually works as far as I could tell. `last_yield` is always set immediately before looping back to `time_since_yield = time.monotonic() - last_yield` so `time_since_yield` will always be `0`.

Reply:
I had convinced myself that there was a reason it does what it does now instead of keeping the interval consistent regardless of loop body, but now I've just confused myself...

In any case it doesn't currently work that way before this request, but I could fix it to.

Reply:
Inverting the yield and the last_yield= lines should fix it.